### ISSUE TITLE
Total cars

### ISSUE THREAD

**Issue by JeyMarsh** [ultralytics/ultralytics#20511]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have a project to count the number of vehicles that enter and leave a parking lot when passing through a line, but I needed the total number of vehicles present by subtracting those that left from those that entered. This project I did only gives the answer in this format: "SolutionResults(classwise_count={'bench': {'IN': 0, 'OUT': 0}}, total_tracks=1". In short, I needed these two numbers "IN and OUT" to become one.

-I am using YOLOv8 and the "cv2" library

### Additional

import os
import cv2
import serial
from ultralytics.solutions import object_counter
from ultralytics import YOLO

model = YOLO(r"Repositório\runs\detect\train12\weights\best.pt")
os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"
cap = cv2.VideoCapture(0)
#Concectando com o Arduino
porta = "COM7"
velocidade = 921600
try:
    conecao = serial.Serial(porta, velocidade, timeout=0.5)
    print("Conectado com a porta", conecao.portstr)
except serial.SerialException:
    print("Não conectada")
    pass

assert cap.isOpened(), "Error reading video file"

# region_points = [(20, 400), (1080, 400)]                                      # line counting
region_points = [(20, 400), (1080, 400), (1080, 360), (20, 360)]  # rectangle region
# region_points = [(20, 400), (1080, 400), (1080, 360), (20, 360), (20, 400)]   # polygon region

# Video writer
w, h, fps = (int(cap.get(x)) for x in (cv2.CAP_PROP_FRAME_WIDTH, cv2.CAP_PROP_FRAME_HEIGHT, cv2.CAP_PROP_FPS))
video_writer = cv2.VideoWriter("object_counting_output.avi", cv2.VideoWriter_fourcc(*"mp4v"), fps, (w, h))

classes_to_count = [0]

# Initialize object counter object
counter = object_counter.ObjectCounter()
counter.set_args(view_img=True,
    show=True,  # display the output
    region=region_points,  # pass region points
    model=(r"Repositório\runs\detect\train12\weights\best.pt"),
     # count specific classes i.e. person and car with COCO pretrained model.
    #tracker="custom_dataset.yaml",
)

# Process video
while cap.isOpened():
    success, im0 = cap.read()

    if not success:
        print("Video frame is empty or processing is complete.")
        break
    tracks = model.track(im0, persist=True, show=False, classes=classes_to_count)
    results = counter(im0)
    resposta = counter(im0)
    print("resposta", resposta)

   # boxes = results[0].boxes.xyxy.cpu().tolist()
   # clss = results[0].boxes.clss.cpu().tolist()
   # confs = results[0].boxes.conf.cpu().tolist()

    #if boxes is not None:
    #    for box, cls, conf in zip(boxes, clss, confs):
    #        color = colors(int(cls), True)

    #    cv2.rectangle(frame, (box[0], box[1]), (box[2], box[3]), color, 2)

    #    label = f'{model.names[int(cls)]} {conf:.2f}'
    #    cv2.putText(im0, label, (int(box[0]), int(box[1]) - 10), cv2.FONT_HERSHEY_SIMPLEX, 0.5, color, 2)

    # Mostra o frame processado com as caixas
   # cv2.imshow("Output", im0)
   # if cv2.waitKey(1) & 0xFF == ord('q'):  # Sai do loop se apertar a tecla 'q' for pressionada
   #     cap.release()
   #     video_writer.release()
   #     cv2.destroyAllWindows()

    # print(results)  # access the output

    video_writer.write(results.plot_im)  # write the processed frame.
    if results == 1:
        print("SolutionResults(classwise_count={'car': {'IN': 0, 'OUT': 0}})")
        conecao.write (b'1')

cv2.Imshow("Output", im0)

cap.release()
video_writer.release()
cv2.destroyAllWindows()  # destroy all opened windows
conecao.close()

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JeyMarsh, thank you for reaching out and for sharing detailed information about your project 🚗🚦! This is an automated response to help guide your next steps while an Ultralytics engineer will also assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for in-depth guides and usage examples, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) workflows. Many common questions are already answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us reproduce and diagnose the issue effectively.

For custom training or model adaptation ❓, please share as much context as possible (dataset examples, training logs, etc.), and make sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for support and collaboration:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for detailed discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for broader community exchange

## Upgrade

To ensure you have the latest features and bug fixes, please upgrade your `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these verified, up-to-date environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for contributing to Ultralytics! 🚀

**Comment by glenn-jocher** [comment]
Hi @JeyMarsh,

To calculate the total vehicles present (IN minus OUT) with the ObjectCounter solution, you can access these values directly from the results object:

```python
# Inside your while loop
results = counter(im0)
in_count = results.in_count
out_count = results.out_count
total_present = in_count - out_count
print(f"Total vehicles present: {total_present}")

# For class-specific counting
# If you want the count for a specific class (e.g., 'car')
if 'car' in results.classwise_counts:
    cars_in = results.classwise_counts['car']['IN']
    cars_out = results.classwise_counts['car']['OUT']
    total_cars_present = cars_in - cars_out
    print(f"Total cars present: {total_cars_present}")
    
    # Send to Arduino if needed
    if total_cars_present > 0:
        conecao.write(b'1')
```

The ObjectCounter returns these counts separately, and you need to calculate the difference yourself to get the total present. You can also access the detailed documentation on the [object counter solution](https://docs.ultralytics.com/reference/solutions/object_counter/) for more options and information.
